### PR TITLE
Friend list retrieval from steam api

### PR DESCRIPTION
**Pull request by megascatterbomb**
Related to https://github.com/MegaAntiCheat/client-backend/pull/88

Utilizes friends list data sent in player objects.

Displays a colour for players in the user's friends list (defaults to dark green)
Displays a colour for players in a cheater's friends list in the same lobby (defaults to white). Also includes an icon that when hovered over lists the cheaters they're friends with.

Adds a setting in the preferences to control friend list retrieval behaviour.

Additional misc changes to help accommodate these changes:
- Changed some default colours to accommodate the new colours and align with precedent set by megascatterbomb.com.
- Fixed icons not grayscaling when player disconnects.
- Icon container width is now dynamic.
- Added a "prod" script to run development mode alongside a separate backend instance.
- Added a dedicated color for convicts.
- Updated fake data to assist with testing these new features.

Future work: Look at making icons toggleable so users can filter out information they don't care about.

### CODE REVIEW

**Comment by Bash-09** [comment]
I'm personally of the opinion that some of this info would be better off represented by icons, and highlight colours be reserved for Verdict/Convictions. Looking at the dev version it's starting to turn into a bit of a rainbow soup and I struggle to remember what all the different colours mean, whereas icons (or even coloured icons, like green and red variations of friends for your friends vs friends of cheaters, for example) might be more informative.

image

This is just a minor complaint though and shouldn't really block the MR, maybe I will move it into an issue to facilitate discussion and allow future action.

**Review by meatmeatmeatmeatmeatmeat — APPROVED** [review]

File: src/components/TF2/Player/playerutils.tsx [hunk 1]
```diff
@@ -78,14 +85,39 @@ function displayColor(
...
+    if (hasCheaterFriendsInLobby(player, cheatersInLobby))
+      return hexToRGB(playerColors['FriendOfCheater'], ALPHA);
+
+    return hexToRGB(playerColors['Player'], ALPHA);
+  }
 
   return hexToRGB(playerColors[verdict], ALPHA);
 }
 
+function getCheatersInLobby(
```
Comment (Bash-09) [inline src/components/TF2/Player/playerutils.tsx:103]: Minor style complaint:
Could probably have a more descriptive name like `getCheaterFriendsInLobby`, that would be more consistent with `hasCheaterFriendsInLobby` and makes it clear that it's not *all* the cheaters in lobby, but just those who are friends of a player.

**Review by Bash-09 — APPROVED** [review]
The behavior works well from my testing, everything looks pretty good to me.

Just had some comments on style and UX, but they are pretty minor.

**Comment by megascatterbomb** [comment]
> I'm personally of the opinion that some of this info would be better off represented by icons, and highlight colours be reserved for Verdict/Convictions. Looking at the dev version it's starting to turn into a bit of a rainbow soup and I struggle to remember what all the different colours mean, whereas icons (or even coloured icons, like green and red variations of friends for your friends vs friends of cheaters, for example) might be more informative. And also because some of these requirements are not mutually exclusive, like someone can be your friend, marked with a verdict, convicted, and be a friend of a cheater, all of which cannot be summarized with one highlight colour.
> 
> image
> 
> This is just a minor complaint though and shouldn't really block the MR, maybe I will move it into an issue to facilitate discussion and allow future action.

Yeah, I mostly stuck to the precedent set by megascatterbomb.com. In the future I plan to make it so users can pick between icons and/or colours for most indicators (or neither), but as long as the code is solid we should be okay going forward.

File: src/components/TF2/Player/playerutils.tsx (hunk 1 above)
Comment (megascatterbomb) [reply src/components/TF2/Player/playerutils.tsx]: Fixed

**Comment by Bash-09** [comment]
> Yeah, I mostly stuck to the precedent set by megascatterbomb.com. In the future I plan to make it so users can pick between icons and/or colours for most indicators (or neither), but as long as the code is solid we should be okay going forward.

Yep sounds good. I for sure don't think it is worth holding anything up for, just something we may want to consider as part of the overall UX going forward.

**Pull request merged**